src/Runner.Worker/ExecutionContext.cs

```diff
@@ -968,6 +968,8 @@ public void QueueAttachFile(string type, string name, string filePath)
             }
 
             _jobServerQueue.QueueFileUpload(_mainTimelineId, _record.Id, type, name, filePath, deleteSource: false);
+
+            _jobServerQueue.QueueResultsUpload(_record.Id, name, filePath, type, deleteSource: false, finalize: true, firstBlock: true, totalLines: 0);
```

Comment:
i would suggest don't change this since you are mixing Results related method into a Actions service method.

Here is what i would suggest:
- In `ExecutionContext`, add `void QueueDiagnosticLogFile(string name, string filePath);` to upload diag log only to Results, similar to how the existing `void QueueSummaryFile(string name, string filePath, Guid stepRecordId);` set up.
- In `DiagnosticLogManager`, call the `ExecutionContext.QueueDiagnosticLogFile()`

This way you get clear isolation for Actions service vs. Results, and easy for us to delete legacy code later.